### ISSUE TITLE
regression: builtin latex rendering doesn't find the right mathematical fonts

### ISSUE THREAD

**Issue by PoldoSbaffini** [matplotlib/matplotlib#6665]
Matplotlib 1.5.1, installed by conda
Python 3.5.1 |Continuum Analytics, Inc.| (default, Jun 15 2016, 15:32:45) 
[GCC 4.4.7 20120313 (Red Hat 4.4.7-1)] on linux

I have this regression, starting from today the matplotlib builtin latex rendering doesn't find the right mathematical fonts.
I have to say that (earlier) today a `conda update --all` had installed a new version of `fontconfig` and on the whole it smells of a packaging problem... anyway...

My code
```
>>> import matplotlib.pyplot as plt
>>> plt.xlabel('$x_1$')
<matplotlib.text.Text object at 0x7fcc8c846c50>
>>> plt.plot(range(3))
[<matplotlib.lines.Line2D object at 0x7fcc7bd69198>]
>>> plt.show()
```
my output, please look at the index of `x`
![figure_1](https://cloud.githubusercontent.com/assets/2244221/16432950/085515bc-3d88-11e6-939c-849a2f51e4c3.png)

**Comment by PoldoSbaffini** [comment]
There is something strange with the font cache,
```
boffi@debian:~$ find | grep fontList.cache
boffi@debian:~$ python
Python 3.5.1 |Continuum Analytics, Inc.| (default, Jun 15 2016, 15:32:45) 
[GCC 4.4.7 20120313 (Red Hat 4.4.7-1)] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> from matplotlib.font_manager import FontManager
>>> FontManager()
/home/boffi/src/miniconda3/lib/python3.5/site-packages/matplotlib/font_manager.py:274: UserWarning: Matplotlib is building the font cache using fc-list. This may take a moment.
  warnings.warn('Matplotlib is building the font cache using fc-list. This may take a moment.')
/home/boffi/src/miniconda3/lib/python3.5/site-packages/matplotlib/font_manager.py:274: UserWarning: Matplotlib is building the font cache using fc-list. This may take a moment.
  warnings.warn('Matplotlib is building the font cache using fc-list. This may take a moment.')
<matplotlib.font_manager.FontManager object at 0x7f22e57b8a58>
>>> 
boffi@debian:~$ find | grep fontList.cache
boffi@debian:~$ 
```

1. at the beginning, I have no `fontList.cache` (it used to be in ~/.cache/matplotlib/fontList.cache`)
2. the font manager tells me (twice, once for TTF and once for AFM, isn't it?) that it's building the font cache and doesn't signal any error
3. at the end, I still cannot see any `font:ist.cache` file...

**Comment by QuLogic** [comment]
It is `fontList.py3k.cache` for Python 3.

**Comment by tacaswell** [comment]
What is in your rcparams?

I can not reproduce this in a fresh environment (`conda create -n mpl151 python=3.5 matplotlib`)

![so](https://cloud.githubusercontent.com/assets/199813/16440237/df12993a-3d8d-11e6-9b9e-72b313015f03.png)

(and yes I know 'works for me' is the worst developer response ever)

**Comment by PoldoSbaffini** [comment]
The `fontList` cache file is in its place, as suggested by QuLogic, under its new name.
Tacaswell asks for my `rcParams`, here it is
[rcParams.txt](https://github.com/matplotlib/matplotlib/files/338841/rcParams.txt)
and shows that in a new environment the subscript is rendered using the correct font,
so I too installed a new environment and yet I have not the correct font.

Is it possible that the new environment is built using the cached components of the installation?.. My problems started following a `conda update --all`.

**Comment by PoldoSbaffini** [comment]
Update

1. I created a new user on my system, had they install a fresh `miniconda3` environment, install `matplotlib` in it, run my example  and everything latex rendering showed up fine... So it's me after all...
2. I removed and reinstalled `miniconda3`, installed `matplotlib` and ran my example — the font cache was not remade, because I let in place all the configuration stuff in hidden files and directories — and voilà the subscript is correct...

I'm sorry that, as I've just realized, I've destroyed the possibility of replying and eventually fixing my particular regression.
